app/src/main/java/org/wikipedia/search/SearchResultsViewModel.kt

```diff
@@ -26,20 +26,20 @@ class SearchResultsViewModel : ViewModel() {
 
     private val batchSize = 20
     private val delayMillis = 200L
-    var resultsCount = mutableListOf<Int>()
+    var resultPairList = mutableListOf<Pair<String, Int>>()
     var searchTerm: String? = null
     var languageCode: String? = null
     lateinit var invokeSource: Constants.InvokeSource
 
     @OptIn(FlowPreview::class) // TODO: revisit if the debounce method changed.
     val searchResultsFlow = Pager(PagingConfig(pageSize = batchSize, initialLoadSize = batchSize)) {
-        SearchResultsPagingSource(searchTerm, languageCode, resultsCount, invokeSource)
+        SearchResultsPagingSource(searchTerm, languageCode, resultPairList, invokeSource)
     }.flow.debounce(delayMillis).cachedIn(viewModelScope)
 
     class SearchResultsPagingSource(
         private val searchTerm: String?,
         private val languageCode: String?,
-        private var resultsCount: MutableList<Int>?,
+        private var resultPairList: MutableList<Pair<String, Int>>,
```

Comment:
Can we give this a clearer name, such as `countsPerLanguageCode`?

Reply:
Done.